plugins/inventory/azure_arc.py

```diff
@@ -0,0 +1,193 @@
+        """
+        Parses the inventory file and populates it.
+        :param loader: an ansible.parsing.dataloader.DataLoader object
+        :param path: the path to the inventory config file
+        """
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        # Load configuration
+        config_data = self._read_config_data(path)
+        subscription_id = config_data.get("subscription_id")
+        resource_group = config_data.get("resource_group")
+        username = config_data.get("username")
+        config_file_path = config_data.get(
+            "config_file_path", "/tmp/azure_arc_ssh_config"
+        )
+        private_key_file = config_data.get(
+            "private_key_file", os.path.expanduser("~/.ssh/id_rsa")
+        )
+
+        if not resource_group:
+            raise ValueError("Azure Resource Group must be provided.")
+        if not username:
+            raise ValueError("Username for SSH connection must be provided.")
+
+        # Try to delete config_file_path, in case it already exists
+        if os.path.exists(config_file_path):
+            os.remove(config_file_path)
+
+        machines = self.get_arc_machines(resource_group, subscription_id)
+
+        for machine in machines:
+            # Generate SSH config using az ssh config
+            new_host_name = self.generate_ssh_config(
+                machine["name"],
+                resource_group,
+                username,
+                private_key_file,
+                config_file_path,
+                subscription_id,
+            )
+            self.inventory.add_host(machine["name"])
+            # Use the generated SSH hostname saved in the config file
+            self.inventory.set_variable(
+                machine["name"], "ansible_host", new_host_name
+            )
+            self.inventory.set_variable(
+                machine["name"], "ansible_connection", "ssh"
+            )
+            # Use the SSH config file path
+            self.inventory.set_variable(
+                machine["name"], "ansible_ssh_common_args", f"-F {config_file_path}"
+            )
+
+    def get_arc_machines(self, resource_group, subscription_id=None):
+        """
+        Retrieve Azure Arc-connected machines using Azure CLI (az connectedmachine list).
+        :param resource_group: Azure Resource Group where Arc machines are located
+        :param subscription_id: Azure Subscription ID (optional)
+        """
+        command = [
+            "az",
+            "connectedmachine",
+            "list",
+            "--resource-group",
+            resource_group,
+            "--output",
+            "json",
+        ]
+
+        if subscription_id is not None:
+            command = command + ["--subscription", subscription_id]
+
+        try:
+            result = subprocess.run(command, capture_output=True, check=True, text=True)
+            machines = json.loads(result.stdout)
+            return [
```

Comment:
Why are you returning a list of dictionaries that each only have one key? Why not just return the list of names?